db/tests/types/test_email.py

```diff
@@ -3,22 +3,35 @@
 from sqlalchemy import text, select, Table, Column, MetaData
 from sqlalchemy.exc import IntegrityError
 from db.types.custom import email
+from db.types.base import PostgresType
 from db.utils import execute_pg_query
 from db.functions.base import ColumnName, Literal, sa_call_sql_function
 from db.functions.operations.apply import apply_db_function_as_filter
 
 
 def test_domain_func_wrapper(engine_with_schema):
     engine, _ = engine_with_schema
-    sel = select(sa_call_sql_function(email.EMAIL_DOMAIN_NAME, text("'test@example.com'")))
+    sel = select(
+        sa_call_sql_function(
+            email.EMAIL_DOMAIN_NAME,
+            text("'test@example.com'"),
+            return_type=PostgresType.TEXT.get_sa_class(engine)
```

Comment:
Is this why you didn't want to use `PostgresType` in `sa_call_sql_function`? There is indeed no good reason why `PostgresType.get_sa_class` should require an engine. I solved this elsewhere... Let me see.

Reply:
There's `db.engine.dummy_engine` for that. See its docstring and its usage in `db/types/operations/convert.py::_get_sa_type_class_id_from_ischema_names`. Here's what you can do:

```diff
diff --git a/db/types/base.py b/db/types/base.py
index d3a65fec..593d1a06 100644
--- a/db/types/base.py
+++ b/db/types/base.py
@@ -4,6 +4,7 @@ from sqlalchemy import text, create_engine as sa_create_engine
 
 from db import constants
 from db.utils import OrderByIds
+from db.engine import get_dummy_engine
 
 
 class DatabaseType(OrderByIds):
@@ -17,13 +18,14 @@ class DatabaseType(OrderByIds):
         """
         return self.value
 
-    def get_sa_class(self, engine):
+    def get_sa_class(self):
         """
         Returns the SA class corresponding to this type or None if this type is not supported by
         provided engine, or if it's ignored (see is_ignored).
         """
+        dummy_engine = get_dummy_engine()
         if not self.is_ignored:
-            ischema_names = engine.dialect.ischema_names
+            ischema_names = dummy_engine.dialect.ischema_names
             return ischema_names.get(self.id)
 
     def is_available(self, engine, type_ids_on_database=None):
```